### PR TITLE
Support loading external xmltv file

### PR DESCRIPTION
**Pull request by rbuehlma**

### CODE REVIEW

File: src/windows.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  return result;
+}
+
+#include <time.h>
+#include <iomanip>
+#include <sstream>
+
+static inline char* strptime(const char* s,
+                          const char* f,
+                          struct tm* tm) {
```
Comment (ksooo) [inline src/windows.h]: Why do you need this? Very ugly.

File: src/client.cpp [hunk 2]
```diff
@@ -201,6 +206,17 @@ ADDON_STATUS ADDON_SetSetting(const char *settingName, const void *settingValue)
       return ADDON_STATUS_NEED_RESTART;
     }
   }
+  
+  if (name == "xmlTVFile")
+  {
+    string xmlTVFileProp = (const char*) settingValue;
```
Comment (ksooo) [inline src/client.cpp]: It is not a good idea (and  not common  practice) to omit the std namespace. This readily can lead to hard to catch errors.

In  c++, we have better cast operators than Java. You should use always use  c++ casts for reasons described a million times in  the net. ;-)

File: src/client.cpp [hunk 3]
```diff
@@ -34,6 +34,7 @@ std::string zatPassword;
 bool zatFavoritesOnly = false;
 bool zatAlternativeEpgService = false;
 bool streamType = false;
+std::string xmlTVFile;
```
Comment (ksooo) [inline src/client.cpp]: No need to define the vars at the beginning of a function. This is not  C. In c++ it is good practice to  define variables where they are first used.

File: src/ZatData.cpp [hunk 4]
```diff
@@ -591,6 +593,9 @@ ZatData::ZatData(const string& u, const string& p, bool favoritesOnly,
   }
   
   ReadDataJson();
+  if (!xmlTVFile.empty()) {
+    xmlTV = new XmlTV(xmlTVFile);
+  }
```
Comment (ksooo) [inline src/ZatData.cpp]: Unlike in Java, in  c++ it is common practice to put  curly braces in the next line.

File: src/ZatData.cpp [hunk 5]
```diff
@@ -529,15 +529,17 @@ int ZatData::GetChannelGroupsAmount()
...
+    const string& xmlTVFile) :
+    recordingEnabled(false), uuid(""), xmlTV(nullptr)
 {
   XBMC->Log(LOG_NOTICE, "Using useragent: %s", user_agent.c_str());
   username = u;
   password = p;
   this->alternativeEpgService = alternativeEpgService;
   this->favoritesOnly = favoritesOnly;
   this->streamType = streamType;
+  this->xmlTVFile = xmlTVFile;
```
Comment (ksooo) [inline src/ZatData.cpp]: Put member init in ctor's initializer list.

**Comment by ksooo** [comment]
Really no offense intended, no doubt the code works, but this is not  c++, it is Java squeezed to get compiled by a c++ compiler. Tbh, it hurts in the eyes to see so many c++ fundamentals missing in the code (reference vs. by value parameters, c style casts, ctor init list vs. init in ctor body,...). 

Sorry, but I just had to say that. Keep up the good work and maybe you are in the mood to learn a bit more  c++?

File: src/windows.h (hunk 1 above)
Comment (rbuehlma) [reply src/windows.h]: strptime is not available on windows. Is there some better way of doing this?

File: pvr.zattoo/addon.xml.in [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <addon id="pvr.zattoo"
-       version="18.0.59"
+       version="18.0.60"
```
[the remark this answers is not in the document]
Comment (rbuehlma) [reply pvr.zattoo/addon.xml.in]: My Idea was to synchronize the major and minor version to Kodi such that it is clear for which version this addon is built. But I guess the minor version is not relevant for compatibility and could therefore be used for feature changes.

**Pull request merged**